### PR TITLE
update to use more null safe packages

### PR DESCRIPTION
**Pull request by jakemac53**
## Description

This updates all the dependencies used in the most basic `test` and `flutter_test` isolates.

## Related Issues

https://github.com/flutter/flutter/issues/53908

## Tests

No tests

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/update_packages.dart
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/update_packages.dart:31]: please use consistent indenting on all lines

**Comment by jakemac53** [comment]
I can't seem to make heads or tails of these failures (can't get any logs?). Some seem to have been some sort of transitive failures potentially (rescheduled jobs?).

**Comment by jakemac53** [comment]
One of the jobs does seem like a legitimate failure, https://cirrus-ci.com/task/5973469030514688?command=main#L943. It seems like another pubspec somewhere in a test fixture needs to allow the latest test package.

**Comment by jakemac53** [comment]
Ah, `flutter create` adds a test dependency of `test: 1.15.3`.

That will need to be updated to something that allows null safety versions, what constraint should I put? (If you leave it up to me it will be `^1.15.0` :P.

**Comment by jakemac53** [comment]
The jobs seem to be running successfully now so i can actually see logs, it looks like we will either be blocked on https://github.com/dart-lang/sdk/commit/6bba75079a1f7166c00ad250a6de6fc3aaff9b17 rolling in from the Dart SDK or  putting in a workaround.

**Review by a14n — APPROVED** [review]

**Comment by jakemac53** [comment]
Ok, it looks like I am waiting for https://github.com/flutter/engine/commits/b955e15ee9c5a8b15664aabb576e01c1233293ad to roll into flutter. The auto-roller appears to have been having a lot of problems though.

I will wait a bit and see if it manages to succeed, I worry about adding the the workarounds because those will end up causing dead code hints in the analyzer once the dart sdk fix rolls in.

**Pull request merged**
